File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        ans, left, right = 0, 0, len(people)-1
        people.sort()
        while left <= right:
            if people[left] + people[right] <= limit:
                left += 1
            right -= 1
            ans += 1
        return ans

File: 917-boats-to-save-people/boats-to-save-people.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
    let ans = 0;
    people.sort((a,b) => a-b);
    let left = 0, right = people.length-1;
    while(left < right) {
        if(people[left] + people[right] <= limit) left++;
        right--;
        ans++;
    }
    if(left === right) ans++;
    return ans;
};